holmes/core/conversations.py

```diff
@@ -262,7 +262,7 @@ def build_issue_chat_messages(
 
 
 def add_or_update_system_prompt(
-    conversation_history: List[Dict[str, str]], ai: ToolCallingLLM, config: Config
+    conversation_history: List[Dict[str, str]], ai: ToolCallingLLM, config: Config, custom_system_prompt: Optional[str] = None
```

Comment:
Possible better to call it `extra_system_prompt` or `additional_system_prompt`, to make it clear you're not overwriting the whole system prompt.